workspaces/jenkins/plugins/scaffolder-backend-module-jenkins/src/config.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { RootConfigService } from '@backstage/backend-plugin-api';
+import Jenkins from 'jenkins';
+
+const JENKINS_CONFIG_KEY: string = 'jenkins';
+const BASE_URL_KEY: string = 'baseUrl';
+const HEADERS_KEY: string = 'headers';
+const USERNAME: string = 'username';
+const API_KEY: string = 'apiKey';
+const CRUMB_ISSUER_ENABLED: string = 'crumbIssuerEnabled';
+
+const HTTP_PREFIX: string = 'http://';
+const HTTP_PROTOCOL: string = 'http://';
+const HTTPS_PROTOCOL: string = 'https://';
+
+export const jenkinsConfigStructure = {
```

Comment:
Think this config should also be added the `config.d.ts` too so that there can be a schema defining what you can configure.

Reply:
Mmm, i don't know if i understood well but it seems is already there below the screenshot of compiled file

![Schermata da 2024-12-06 13-04-29](https://github.com/user-attachments/assets/b249a8b4-a345-45da-8969-b5b06ae25888)

Reply:
so basically copy-paste this file https://github.com/backstage/community-plugins/blob/main/workspaces/jenkins/plugins/jenkins-backend/config.d.ts
and stripping out the config you don't need, and hook it up in your `package.json`.

This was missing in the current implementation, which causes issues if adopters have the jenkins module installed without having the `jenkins-backend` plugin installed.